packages/web/src/scss/tools/_list.scss

```diff
@@ -4,12 +4,17 @@
 // Example: to-string((1, 2, 3), '-') will return '1-2-3'
 @function to-string($list, $separator: ' ') {
     $string: '';
+    $first: true;
 
     @for $i from 1 through list.length($list) {
-        $string: $string + list.nth($list, $i);
+        $item: list.nth($list, $i);
 
-        @if $i < list.length($list) {
-            $string: $string + $separator;
+        @if $item != null and $item != '' {
+            @if not $first {
```

Comment:
imho you can check if the `string` is not ''. And it would do the same and you don't need another variable.